### PR TITLE
Robin/offscreen rendering class

### PR DESCRIPTION
**Pull request by robin-imverse**
Moved ImVkOffscreenPass as an independant class with more controls during creation.

### CODE REVIEW

File: src/ImVkTextureAttachment.cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ImVkTextureAttachment::ImVkTextureAttachment() : 
+  m_image{VK_NULL_HANDLE}
+  , m_memory{VK_NULL_HANDLE}
+  , m_view{VK_NULL_HANDLE}
+  , m_sampler{VK_NULL_HANDLE}
+  , m_descriptor{VK_NULL_HANDLE, VK_NULL_HANDLE, VK_IMAGE_LAYOUT_UNDEFINED}
+  , m_format{VK_FORMAT_UNDEFINED}
+  , m_isImageOwner{false}
+  , m_isImageViewOwner{false}
+  , m_clearFlag{VK_ATTACHMENT_LOAD_OP_CLEAR}
```
Comment (ImCRV) [inline src/ImVkTextureAttachment.cpp:12]: I cannot see this member in the header...

Comment (ImCRV) [reply src/ImVkTextureAttachment.cpp:12]: (m_clearFlag and m_storeFlag)

File: src/ImVkTextureAttachment.hpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#pragma once
```
Comment (ImCRV) [inline src/ImVkTextureAttachment.hpp:1]: This file seems to be an unused duplicate

File: src/ImVkTextureAttachment.cpp (hunk 1 above)
Comment (robin-imverse) [reply src/ImVkTextureAttachment.cpp:12]: weird, I have it here and I pushed everything

Comment (robin-imverse) [reply src/ImVkTextureAttachment.cpp:12]: done, it was a duplicated header. I deleted it and pushed.

**Review by ImCRV — APPROVED** [review]

**Pull request merged**